### ISSUE TITLE
How to read raw flash?

### ISSUE THREAD

**Issue by TediumRemedy** [texane/stlink#644]
Is there a way to read raw flash bytes, for example, if I am saving some config data to the last page of the flash, and later would like to examine it how it changed? Something similar to what ST's STLink utility for Windows does - showing whole flash in hex.
Thanks

**Comment by xor-gate** [comment]
You need st-flash tool which does read/writing of STM32 flash.

```
stlinkv2 command line: ./st-flash [--debug] [--reset] [--serial <serial>] [--format <format>] [--flash=<fsize>] {read|write} <path> <addr> <size>
```

Probably do something like (untested):

```
st-flash --format binary read mydump.bin 0x08000000 0x100
```

**Comment by Nightwalker-87** [comment]
Feedback closes #644.
